dune-project

```diff
@@ -21,6 +21,7 @@
  (depends
   (ocaml
    (>= 4.02.3))
+  (dune (or (and :with-test (>= 2.8)) (>= 2.2)))
```

Comment:
I think the more idiomatic way to express that is:

```
(dune (>= 2.2))
(dune (and (>= 2.8) :with-test))
```

Reply:
also, doesn't that split the universe? released version requires dune < 2.8 in CI and unreleased one requires >= 2.8

Reply:
Duplicating atoms feels extremely odd to me and even though it will work, I feel like it relies a bit too much on the implementation details of opam for me to feel comfortable expressing the dependency this way.

Reply:
Only the tests are impacted, released version is co-installable with all dune version >= 2.2 and so would be the next one!

Reply:
you can find the "duplicated" pattern in several places in opam-repository: `git grep -e 'ocaml".*with-test.*>' -e 'ocaml".*>.*with-test'` (vchan, ppx_bitstring, ocamlformat, conjury)

@kit-ty-kate what is the preferred way to do that? (have different bounds for a dependency depending on `:with-test`)